exchanges/btcmarkets/btcmarkets.go

```diff
@@ -379,52 +536,55 @@ func (b *BTCMarkets) SendHTTPRequest(path string, result interface{}) error {
 }
 
 // SendAuthenticatedRequest sends an authenticated HTTP request
-func (b *BTCMarkets) SendAuthenticatedRequest(reqType, path string, data, result interface{}) (err error) {
+func (b *BTCMarkets) SendAuthenticatedRequest(method, path string, data interface{}, result interface{}) (err error) {
 	if !b.AllowAuthenticatedRequest() {
 		return fmt.Errorf(exchange.WarningAuthenticatedRequestWithoutCredentialsSet,
 			b.Name)
 	}
 
-	n := b.Requester.GetNonce(true).String()[0:13]
-
-	var req string
-	payload := []byte("")
+	strTime := strconv.FormatInt(time.Now().UTC().UnixNano()/1000000, 10)
 
-	if data != nil {
+	var body io.Reader
+	var payload, hmac []byte
+	switch data.(type) {
+	case map[string]interface{}:
 		payload, err = common.JSONEncode(data)
 		if err != nil {
 			return err
 		}
-		req = path + "\n" + n + "\n" + string(payload)
-	} else {
-		req = path + "\n" + n + "\n"
-	}
-
-	hmac := crypto.GetHMAC(crypto.HashSHA512,
-		[]byte(req), []byte(b.API.Credentials.Secret))
-
-	if b.Verbose {
-		log.Debugf(log.ExchangeSys, "Sending %s request to URL %s with params %s\n",
-			reqType,
-			b.API.Endpoints.URL+path,
-			req)
+		body = bytes.NewBuffer(payload)
+		strMsg := method + btcMarketsAPIVersion + path + strTime + string(payload)
+		hmac = crypto.GetHMAC(crypto.HashSHA512,
+			[]byte(strMsg), []byte(b.API.Credentials.Secret))
+	case []interface{}:
```

Comment:
Since you're doing the same logic in both case statements, you can do "case map[string]interface{}, []interface{}:"